pontoon/base/forms.py

```diff
@@ -97,11 +99,36 @@ def assign_users_to_groups(self, group_name, users):
 
         group.user_set.clear()
 
+        before_count = self.user.badges_promoted_users
+        now = timezone.now()
+
         if users:
             group.user_set.add(*users)
 
         log_group_members(self.user, group, (add_users, remove_users))
 
+        after_count = self.user.badges_promoted_users
+
+        # Check if user was demoted from Manager to Translator
+        # In this case, it doesn't count as a promotion
```

Comment:
Given that this code seems to be the only consumer of the `PermissionChangelog` model, we should refactor it to make retrieving promotions more straightforward. At least we should add a TODO explaining that and reference https://github.com/mozilla/pontoon/issues/2195.